cache/demux_test.go

```diff
@@ -24,6 +24,202 @@ import (
 	clientv3 "go.etcd.io/etcd/client/v3"
 )
 
+func TestRangeTracking(t *testing.T) {
+	tests := []struct {
+		name      string
+		capacity  int
+		initRev   int64
+		eventRevs []int64
+		wantMin   int64
+		wantMax   int64
+	}{
+		{
+			name:      "history not full",
+			capacity:  2,
+			initRev:   1,
+			eventRevs: []int64{1},
+			wantMin:   1,
+			wantMax:   1,
+		},
+		{
+			name:      "history at exact capacity",
+			capacity:  2,
+			initRev:   1,
+			eventRevs: []int64{1, 2},
+			wantMin:   1,
+			wantMax:   2,
+		},
+		{
+			name:      "history overflow with eviction",
+			capacity:  2,
+			initRev:   1,
+			eventRevs: []int64{1, 2, 3},
+			wantMin:   2,
+			wantMax:   3,
+		},
```

Comment:
```suggestion
		},
		{
			name:      "history overflow not continuous",
			capacity:  2,
			initRev:   2,
			eventRevs: []int64{4, 8, 16},
			wantMin:   5,
			wantMax:   16,
		},
```

Please remember that revisions are not guaranteed to be continious.